src/ksym.rs

```diff
@@ -91,29 +95,25 @@ impl KSymResolver {
         }
     }
 
-    pub fn find_address_ksym(&self, addr: u64) -> Option<&Ksym> {
+    pub fn find_addresses_ksym(&self, addr: u64) -> impl Iterator<Item = &Ksym> {
         let mut l = 0;
         let mut r = self.syms.len();
 
-        if !self.syms.is_empty() && self.syms[0].addr > addr {
-            return None;
-        }
-
-        while l < (r - 1) {
-            let v = (l + r) / 2;
-            let sym = &self.syms[v];
+        while l < r {
+            let m = (l + r) / 2;
+            let sym = &self.syms[m];
 
-            if sym.addr == addr {
-                return Some(sym);
-            }
             if addr < sym.addr {
-                r = v;
+                r = m;
             } else {
-                l = v;
+                l = m + 1;
```

Comment:
I think this is wrong. `r = m` means that r is supposed to be the correct index at all times (except for cases when there is no element that satisfies condition, which is when all syms have sym.addr > addr; but this corner case is handled properly in the loop after this binary search (at least in my example). But `addr < sym.addr` is not the condition we are looking for, we want to find rightmost element that satisfies addr >= sym.addr.

Please take a look at my examples from the discussion before. I think they are more correct (though there could still be bugs I missed, but let's review that).